### PR TITLE
DDF-2802 add jbake to documentation process

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Prototypes using jbake to create automatically-generated documentation based on headers/front-matter. 

#### Who is reviewing it?

@mcalcote @vina @jrnorth @Lambeaux @oconnormi 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Build](https://github.com/orgs/codice/teams/build)
[Docs](https://github.com/orgs/codice/teams/docs)

#### Choose 2 committers to review/merge the PR.

@pklinef
@shaundmorris
@stustison

#### How should this be tested? (List steps with links to updated documentation)

- fresh build with a clean repo
- no test failures or `unresolved directive` errors in rendered documentation
- all marked content included correctly

#### Any background context you want to provide?

Some sections are still hard-coded if: 
- they will rarely change (structurally, even if content is revised)
- they rely heavily on a very specific order of placement in the finished product
- they are self-contained in a single file/document.

Another consideration is that some sections may still need further refinement.

As a prototype, many sections were not included to simplify the prototype, but I tried to make the structure complex enough to prove concepts.

When building, there may be warnings in the logs about headers, as only files that required headers to be included were updated with headers. These will need to be added to all `contents.adoc` files in future iterations. 

#### What are the relevant tickets?

[DDF-2802](https://codice.atlassian.net/browse/DDF-2802)

#### Screenshots (if appropriate)

[ddf-2802.zip](https://github.com/codice/ddf/files/817569/ddf-2802.zip)

#### Checklist:
- [x] Documentation Updated!
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: distribution/docs/src/main/jdocs/templates/documentation.ftl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+= ${branding-expanded} Documentation
+include::${project.build.directory}/doc-contents/_contents/jconfig.adoc[]
+
+
+:sectnums!:
+== Introduction
+:sectnums:
+
+include::{adoc-include}/_introduction/introduction-contents.adoc[]
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/templates/documentation.ftl]: This is a file that is unlikely to be renamed/split further and is context-dependent regarding placement, so it is hard-coded/fixed.

File: distribution/docs/src/main/jdocs/templates/documentation.ftl [hunk 2]
```diff
@@ -0,0 +1,31 @@
+= ${branding-expanded} Documentation
+include::${project.build.directory}/doc-contents/_contents/jconfig.adoc[]
+
+
+:sectnums!:
+== Introduction
+:sectnums:
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/templates/documentation.ftl]: Potentially, these major sections could be separated into their own templates, which would clean up what appears at this top-level.

File: distribution/docs/src/main/resources/_contents/_endpoints/catalog-rest-endpoint-contents.adoc [hunk 3]
```diff
@@ -1,3 +1,9 @@
+:title: Catalog REST Endpoint
+:type: endpoint
+:status: published
+:filename: catalog-rest-endpoint-contents.adoc
+:link: _catalog_rest_endpoint
+:summary: allows clients to perform CRUD operations on the Catalog using REST, a simple architectural style that performs communication using HTTP.
 
 ==== Catalog REST Endpoint
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_endpoints/catalog-rest-endpoint-contents.adoc]: Need to remove this line as the template creates a section heading from the title property declared on line 1.

File: distribution/docs/src/main/resources/_contents/jconfig.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+:toc: left
+:example-caption:
+:source-highlighter: coderay
+:sectlinks:
+:sectanchors:
+:data-uri:
+:sectnums:
+:branding: DDF
+:download-url: https://github.com/codice/ddf/releases
+:adoc-include: ${project.build.directory}/doc-contents/content
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/jconfig.adoc]: to keep from colliding with the existing docs, many of these files have been prepended with a `j`. This path is slightly different than in the regular `config.adoc`. These would eventually merge once the static docs were fully replaced by the automated ones.

File: distribution/docs/src/main/resources/configuring-managed-service-factory-bundles-contents.adoc [hunk 5]
```diff
@@ -1,99 +0,0 @@
-
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/configuring-managed-service-factory-bundles-contents.adoc]: this file was a copy error. it should be removed from other branches as well. (The _real_ one is in `src/main/resources/_contents`)

File: distribution/docs/src/main/jdocs/content/_endpoints/endpoints-intro-contents.adoc [hunk 6]
```diff
@@ -0,0 +1,28 @@
+
+.Endpoints
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_endpoints/endpoints-intro-contents.adoc]: These section titles appear really similar to the titles above and might need some indentation as well.
screen shot 2017-03-14 at 9 46 18 am

**Comment by mcalcote** [comment]
The alignment/justification on the blocks of text sometimes is odd. Not related to the changes here, but maybe we should have a line break when urls are present, in general.
 
screen shot 2017-03-14 at 9 53 22 am

File: distribution/docs/src/main/jdocs/content/_endpoints/endpoints-intro-contents.adoc (hunk 6 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_endpoints/endpoints-intro-contents.adoc]: True. In the full version, there is a block of intro text between the titles.

File: distribution/docs/src/main/jdocs/jbake.properties [hunk 7]
```diff
@@ -0,0 +1,11 @@
+site.host=http://jbake.org
```
Comment (pklinef) [inline distribution/docs/src/main/jdocs/jbake.properties]: Should this be http://codice.org/ddf/?

Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/jbake.properties]: probably. It ultimately doesn't matter as the jbake process doesn't produce a viable site to host, just an asciidoc page to be rendered. If it'd let me remove it altogether, I would.

File: distribution/docs/src/main/jdocs/content/_endpoints/cometd-endpoint-contents.adoc [hunk 8]
```diff
@@ -0,0 +1,541 @@
...
+
+===== Configuring CometD Endpoint
+
+Configure the CometD endpoint from the ${admin-console}:
+
+. Navigate to the *${admin-console}*.
+. Select *${ddf-ui}* application.
+. Select *Configuration* tab.
+
+include::{adoc-include}/_tables/org.codice.ddf.ui.search.standard.endpoint-table-contents.adoc[]
```
Comment (shaundmorris) [inline distribution/docs/src/main/jdocs/content/_endpoints/cometd-endpoint-contents.adoc]: Saw this error here: `Unresolved directive in content/_endpoints/cometd-endpoint-contents.adoc - include::/Users/rlarsen/Builds/DDF/ddf/distribution/docs/target/doc-contents/content/_tables/org.codice.ddf.ui.search.standard.endpoint-table-contents.adoc[]`

Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_endpoints/cometd-endpoint-contents.adoc]: fixed.

**Comment by ricklarsen** [comment]
Simplified this PR to show infrastructure changes. Included minimal content to satisfy template.

File: distribution/docs/src/main/resources/_contents/_endpoints/catalog-rest-endpoint-contents.adoc [hunk 9]
```diff
@@ -1,3 +1,9 @@
+:title: Catalog REST Endpoint
+:type: endpoint
+:status: published
+:filename: catalog-rest-endpoint-contents.adoc
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_endpoints/catalog-rest-endpoint-contents.adoc]: Can you use `content.file` instead?

http://jbake.org/docs/2.5.1/#page_post_custom

Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_endpoints/catalog-rest-endpoint-contents.adoc]: Fixed.

**Pull request merged**